src/aggregator/aggregator/aggregator_test.go

```diff
@@ -416,37 +416,95 @@ func TestAggregatorAddUntimedSuccessNoPlacementUpdate(t *testing.T) {
 	require.Equal(t, 1, len(agg.shards[1].metricMap.entries))
 }
 
+//nolint: dupl
 func TestAggregatorAddUntimedToTimed(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
 	agg, _ := testAggregator(t, ctrl)
 	agg.opts = agg.opts.SetTimedForResendEnabled(true)
 	metas := metadata.StagedMetadatas{testStagedMetadatas[0]}
-	metas[0].Pipelines = append(metas[0].Pipelines, metas[0].Pipelines[0])
+	// add another pipeline
```

Comment:
the tests were kind of a mess before. previously the test data had 2 pipelines, but the pipelines were same aggregationKey. this test worked because timed and untimed creation different entries in the map, because timed is part of the entry key.  so the test was pretty misleading and didn't work as expected when copying to the new test below.

Reply:
and aggregationKey deduping is scoped to a single entry